library/core/resource-loader/src/main/java/org/quiltmc/qsl/resource/loader/mixin/DefaultResourcePackMixin.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.resource.loader.mixin;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.function.Predicate;
+
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.resource.DefaultResourcePack;
+import net.minecraft.resource.ResourceType;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.resource.loader.impl.ModNioResourcePack;
+import org.quiltmc.qsl.resource.loader.impl.ResourceLoaderImpl;
+
+@Mixin(DefaultResourcePack.class)
+public abstract class DefaultResourcePackMixin {
+	// Redirects all resource access to the MC resource pack.
+	@Unique
+	final ModNioResourcePack internalPack = this.locateAndLoad();
+
+	@SuppressWarnings({"ConstantConditions", "EqualsBetweenInconvertibleTypes"})
+	@Unique
+	private ModNioResourcePack locateAndLoad() {
+		return ResourceLoaderImpl.locateAndLoadDefaultResourcePack(
+				this.getClass().equals(DefaultResourcePack.class) ?
+						ResourceType.SERVER_DATA : ResourceType.CLIENT_RESOURCES
+		);
+	}
+
+	/**
+	 * @author QuiltMC
+	 * @reason Rewrite default resource access to avoid resource leaking.
+	 */
+	@Overwrite
```

Comment:
Are these overwrites to fix that eclipse classpath bug? Or is there more context. If this fixes classpath for eclipse then do mention it in class docs so people understand the purpose of these.

Reply:
I mean, the reason field does mention that this is to fix the resource leaking issue.

Reply:
The bug is mentioned on the class javadoc.